StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/SA1633CodeFixProvider.cs

```diff
@@ -0,0 +1,84 @@
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = nameof(SA1633CodeFixProvider))]
+    [Shared]
+    public class SA1633CodeFixProvider : CodeFixProvider
+    {
+        private static readonly ImmutableArray<string> FixableDiagnostics =
+            ImmutableArray.Create(FileHeaderAnalyzers.SA1633DescriptorMissing.Id);
+
+        /// <inheritdoc/>
+        public override ImmutableArray<string> FixableDiagnosticIds => FixableDiagnostics;
+
+        /// <inheritdoc/>
+        public override FixAllProvider GetFixAllProvider()
+        {
+            return CustomFixAllProviders.BatchFixer;
+        }
+
+        /// <inheritdoc/>
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            foreach (Diagnostic diagnostic in context.Diagnostics.Where(d => FixableDiagnostics.Contains(d.Id)))
+            {
+                context.RegisterCodeFix(CodeAction.Create(DocumentationResources.SA1633CodeFix, token => GetTransformedDocumentAsync(context.Document, token), equivalenceKey: nameof(SA1633CodeFixProvider)), diagnostic);
+            }
+
+            return SpecializedTasks.CompletedTask;
+        }
+
+        private static async Task<Document> GetTransformedDocumentAsync(Document document, CancellationToken cancellationToken)
+        {
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var settings = document.Project.AnalyzerOptions.GetStyleCopSettings();
+
+            var fileHeader = FileHeaderHelpers.ParseFileHeader(root);
+            var newSyntaxRoot = fileHeader.IsMissing ? AddHeader(root, document.Name, settings) : ReplaceHeader(document, root, settings);
+
+            return document.WithSyntaxRoot(newSyntaxRoot);
+        }
+
+        private static SyntaxNode ReplaceHeader(Document document, SyntaxNode root, StyleCopSettings settings)
+        {
+            var existingTrivia = root.GetLeadingTrivia().Where(x => !x.IsKind(SyntaxKind.EndOfLineTrivia) && !x.IsKind(SyntaxKind.SingleLineCommentTrivia));
+            return root.WithLeadingTrivia(CreateNewHeader(document.Name, settings).AddRange(existingTrivia).Add(SyntaxFactory.CarriageReturnLineFeed));
+        }
+
+        private static SyntaxNode AddHeader(SyntaxNode root, string name, StyleCopSettings settings)
+        {
+            var newTrivia = CreateNewHeader(name, settings).AddRange(root.GetLeadingTrivia());
+            if (!newTrivia.Last().IsKind(SyntaxKind.EndOfLineTrivia))
+            {
+                newTrivia = newTrivia.Add(SyntaxFactory.CarriageReturnLineFeed);
+            }
+
+            return root.WithLeadingTrivia(newTrivia);
+        }
+
+        private static SyntaxTriviaList CreateNewHeader(string filename, StyleCopSettings settings)
+        {
+            return SyntaxFactory.ParseLeadingTrivia($@"// <copyright file=""{filename}"" company=""{settings.DocumentationRules.CompanyName}"">
+//   {settings.DocumentationRules.CopyrightText}
```

Comment:
:exclamation: The first and last lines should only be included if `settings.DocumentationRules.XmlHeader` is true.

:exclamation: There should be exactly one space between `//` and the copyright text. Users can add spaces to the custom copyright text if they want additional indentation.

:exclamation: The copyright text may have multiple lines. You need to split on `\n` and add `// ` to the beginning of each line (but don't add the space before blank lines).